thewalrus/_torontonian.py

```diff
@@ -147,7 +148,7 @@ def numba_tor(O):  # pragma: no cover
         kept_rows = np.concatenate((kept_modes, kept_modes + N))
         O_XX = nb_ix(O, kept_rows, kept_rows)
 
-        bottom = np.sqrt(np.linalg.det(I - O_XX))
+        bottom = np.sqrt(O.dtype.type(np.real(np.linalg.det(I - O_XX))))

```

Comment:
I'm assuming `np.linalg.det(I - O_XX)` will always be close to real based on the other discussions in this PR. This change is thus only here to solve some instability issues? If the real part should only show up if the complex part is small, then there's also the `np.real_if_close` function.

Reply:
Wow I didn't know about the `real_if_close` function, I was in fact seeking out something like this.  But I think we don't want to make it conditional even, since we are making an input domain restriction.  But I certainly missed this.  All determinates should have imaginary parts dropped.  And recursive formula achieve the same by making the Cholesky diagonal real.